app/models/invite_redeemer.rb

```diff
@@ -49,11 +55,11 @@ def invited_user
   end
 
   def process_invitation
+    approve_account_if_needed
```

Comment:
If we don't move it up this test fails:

rspec ./spec/models/invite_redeemer_spec.rb:55 # InviteRedeemer#redeem should redeem the invite if invited by non staff but not approve

because the user gets created and then it does the approve on the user model but the user model doesn't check for the invited by staff.